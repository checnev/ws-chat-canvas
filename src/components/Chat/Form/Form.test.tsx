import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";
import pretty from 'pretty';
import { Formik, FormikProps } from 'formik';
import ChatForm, { ChatFormValues } from 'components/Chat/Form';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const setUp = () => (
  <Formik
    initialValues={{
      text: '',
    }}
    onSubmit={(values: ChatFormValues) => {
    }}
  >
    {(props: FormikProps<ChatFormValues>) => <ChatForm {...props} />}
  </Formik>
);

it('should render ChatForm', () => {
  act(() => {
    render(setUp(), container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
