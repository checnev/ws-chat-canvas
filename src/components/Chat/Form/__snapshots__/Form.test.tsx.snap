// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChatForm 1`] = `"<form action=\\"#\\" class=\\"chatForm\\"><textarea name=\\"text\\" placeholder=\\"What's new ?\\"></textarea><button type=\\"submit\\" class=\\"chatForm__submit\\"></button></form>"`;
